lib/core/test/unit/Cardano/Wallet/DB/Properties.hs

```diff
@@ -363,13 +385,16 @@ prop_readAfterPut putOp readOp db (ShowFmt key, ShowFmt a) =
     monadicIO (setup >> prop)
   where
     setup = do
-        liftIO (cleanDB db)
-        (cp, meta) <- pick arbitrary
-        liftIO $ unsafeRunExceptT $ createWallet db key cp meta mempty
-    prop = liftIO $ do
-        unsafeRunExceptT $ putOp db key a
-        res <- readOp db key
-        ShowFmt res `shouldBe` ShowFmt (pure a)
+        run $ cleanDB db
+        (InitialCheckpoint cp, meta) <- namedPick "Initial Checkpoint" arbitrary
+        run $ unsafeRunExceptT $ createWallet db key cp meta mempty
+    prop = do
+        run $ unsafeRunExceptT $ putOp db key a
+        res <- run $ readOp db key
+        let fa = pure a
+        monitor $ counterexample $ "\nInserted\n" <> pretty fa
+        monitor $ counterexample $ "\nRead\n" <> pretty res
+        assertWith "Inserted == Read" (res == fa)
```

Comment:
reworked using `monitor` and `run` to get better counter examples printed on failures. Did this for debugging and kept it as it produces a nicer failure.